openfisca_aotearoa/variables/demographics/demographics.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+
+# Import from openfisca-core the common python objects used to code the legislation in OpenFisca
+from openfisca_core.model_api import *
+# Import the entities specifically defined for this tax and benefit system
+from openfisca_aotearoa.entities import Person, Family
+
+
+# This variable is a pure input: it doesn't have a formula
+class date_of_birth(Variable):
+    value_type = date
+    default_value = date(1970, 1, 1)  # By default, if no value is set for a simulation, we consider the people involved in a simulation to be born on the 1st of Jan 1970.
+    entity = Person
+    label = u"Birth date"
+    definition_period = ETERNITY  # This variable cannot change over time.
+    reference = u"https://en.wiktionary.org/wiki/birthdate"
+
+
+# This variable is a pure input: it doesn't have a formula
+class due_date_of_birth(Variable):
+    value_type = date
+    default_value = date(1970, 1, 1)  # By default, if no value is set for a simulation, we consider the people involved in a simulation to be born on the 1st of Jan 1970.
+    entity = Person
+    label = u"Birth due date"
+    definition_period = ETERNITY  # This variable cannot change over time.
+    reference = u""
+
+
+class age(Variable):
+    value_type = int
+    entity = Person
+    definition_period = MONTH
+    label = u"The age of a Person (in years)"
+    # A person's age is computed according to their birth date.
+
+    def formula(persons, period, parameters):
+        birth = persons('date_of_birth', period)
+        birth_year = birth.astype('datetime64[Y]').astype(int) + 1970
+        birth_month = birth.astype('datetime64[M]').astype(int) % 12 + 1
+        birth_day = (birth - birth.astype('datetime64[M]') + 1).astype(int)
+
+        is_birthday_past = (birth_month <= period.start.month) + (birth_month == period.start.month) * (birth_day <= period.start.day)
+        is_birthday_past = (birth_month < period.start.month) + (birth_month == period.start.month) * (birth_day <= period.start.day)
+        return (period.start.year - birth_year) - where(is_birthday_past, 0, 1)  # If the birthday is not passed this year, substract one year
```

Comment:
> With the addition of `due_date_of_birth`, not all Persons can have an `age`, which can be counterintuitive.
> I don't think this is a major problem, but I want to make it explicit.

Reply:
The age is computed as -1 which is arguably still an age. I considered modifying the age variable to aways be a whole number but I believe -1 is more precise.